===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(recutl_parse_db_from_file:parser->source)) at 22 (line 251, column 11),
        Insert: if (true) free(*(recutl_parse_db_from_file:parser)) at 26 (line 253, column 15) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/utils/recutl.c
+++ b/table1/benchmarks/recutils-1.8/utils/recutl.c
@@ -250,11 +250,13 @@ recutl_parse_db_from_file (FILE *in,
       rset_type = rec_rset_type (rset);
       if (rec_db_type_p (db, rset_type))
         {
+          free(parser->source);
           if (rset_type)
             recutl_fatal (_("duplicated record set '%s' from %s.\n"),
                           rset_type, file_name);
           else
             {
+              free(parser);
               /* Special case: the database already contains anonymous
                  records (with no type) and the record set to be
                  inserted also contains anonyous records.  In this
